### PR TITLE
Replace ip/prefix map caches with btree caches.

### PR DESCRIPTION
**Pull request by taktv6**

### CODE REVIEW

File: net/prefix_cache.go [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
-import "sync"
+import (
+	"sync"
+
+	"github.com/google/btree"
+)
 
 const (
-	prefixCacheInitialSize = 1000000
+	prefixCacheBTreeGrade = 3500
```
Comment (BarbarossaTM) [inline net/prefix_cache.go]: why 3500?

File: net/prefix_cache.go [hunk 2]
```diff
@@ -15,25 +19,26 @@ func init() {
...
 type pfxCache struct {
-	cache   map[Prefix]*Prefix
 	cacheMu sync.Mutex
+	tree    *btree.BTree
 }
 
 func newPfxCache() *pfxCache {
 	return &pfxCache{
-		cache: make(map[Prefix]*Prefix, prefixCacheInitialSize),
+		tree: btree.New(3500),
```
Comment (BarbarossaTM) [inline net/prefix_cache.go]: why 3500?

File: net/ip_cache.go [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
-import "sync"
+import (
+	"sync"
+
+	"github.com/google/btree"
+)
 
 const (
-	ipCacheInitialSize = 1000000
+	ipCacheBTreeGrade = 3500
```
Comment (BarbarossaTM) [inline net/ip_cache.go]: why 3500?

File: net/prefix_cache.go (hunk 1 above)
Comment (taktv6) [reply net/prefix_cache.go]: Because it's very good!
Seriously: I've benchmarked this to be fastest on my machine :D

File: net/ip_cache.go (hunk 3 above)
Comment (taktv6) [reply net/ip_cache.go]: Because it's very good!
Seriously: I've benchmarked this to be fastest on my machine :D

**Pull request merged**